### PR TITLE
#22 Allow audit to use ossi credentials

### PR DESCRIPTION
**Pull request by klehrke-sonatype**
Allows the user to use their ossi credentials if present in their config file.

### CODE REVIEW

File: config/configuration.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ import (
 )
 
 type OSSIConfig struct {
-	Username string `yaml:Username`
-	Password string `yaml:Password`
+	Username string `yaml:"Username"`
+	Token string `yaml:"Token"`
```
Comment (klehrke-sonatype) [inline config/configuration.go:28]: Oops! this is suppose to be token!

File: audit/audit.go [hunk 2]
```diff
@@ -28,14 +29,26 @@ import (
...
-func ProcessPaths(libPaths []string, libs []string, files []string) (count int) {
+type Audit struct {
+	OssiConfig         config.OSSIConfig
+	HasProperOSSICreds bool
+}
+
+func New(ossiConfig config.OSSIConfig) *Audit {
+    return &Audit{
+        OssiConfig:         ossiConfig,
+        HasProperOSSICreds: len(ossiConfig.Username) > 0 && len(ossiConfig.Token) > 0,
```
Comment (DarthHater) [inline audit/audit.go:40]: This is really cool, but you could LIKELY do it as a method hanging off Audit itself

File: linker/linker_test.go [hunk 3]
```diff
@@ -31,7 +34,8 @@ func TestDoLink(t *testing.T) {
 	// FIXME: By default we are returned the number of vulnerable packages. We
 	// might want to add a way to get the total number of packages as well.
 	var expected = 0
-	countFromSesameStreet := DoLink(args)
+	myLinker := New(config.OSSIConfig{})
+	countFromSesameStreet := myLinker.DoLink(args)
```
Comment (DarthHater) [inline linker/linker_test.go:38]: Glad to see the count is still kicking

File: audit/audit.go (hunk 2 above)
Comment (klehrke-sonatype) [reply audit/audit.go:40]: this was an artifact of testing as well :(

**Review by ken-duck — APPROVED** [review]
LGTM

File: audit/audit.go [hunk 4]
```diff
@@ -52,13 +67,27 @@ func AuditBom(deps []packageurl.PackageURL) (count int) {
...
-	ossi := ossindex.New(
-		logger.GetLogger(),
-		types.Options{
-			Version:     "development",
-			Tool:        "cheque",
-			DBCacheName: "cheque-cache",
-		})
+	var ossi *ossindex.Server
+
+    if a.HasProperOssiCredentials() {
```
Comment (DarthHater) [inline audit/audit.go:72]: You might dig this:

```
	options := types.Options{Version: "development", Tool: "cheque", DBCacheName: "cheque-cache"}
	if a.HasProperOssiCredentials() {
		options.Username = a.OssiConfig.Username
		options.Token = a.OssiConfig.Token
	} 
	ossi := ossindex.New(
		logger.GetLogger(),
		options
		})
```

Comment (klehrke-sonatype) [reply audit/audit.go]: done!

**Review by DarthHater — APPROVED** [review]

**Pull request merged**